drivers/gpio/Kconfig.altera

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config GPIO_ALTERA_PIO
+	bool "Altera PIO"
+	default y
+	depends on DT_HAS_ALTR_PIO_1_0_ENABLED
+	help
+	  Enable driver for Altera PIO
+
+config GPIO_ALTERA_PIO_OUTSET
+	bool "Outset Register"
+	help
+	  Enable outset register to specify which bit of the output port to set.
+
+config GPIO_ALTERA_PIO_OUTCLEAR
+	bool "Outclear Register"
+	help
+	  Enable outclear register to specify which output bit to clear.
```

Comment:
These sound more like properties of the synthesized hardware? Should they go in the DTS instead?

Reply:
These are the features for IP specified. It is configured here to reduce the code footprint during the runtime.

Reply:
DT properties are converted into compile-time const items, so theoretically they should work equally well as Kconfig.

The mantra is DT for configuring hardware snd Kconfig for configuring software.